packages/smart-contracts/scripts-create2/xdeployer.ts

```diff
@@ -0,0 +1,163 @@
+import { HardhatRuntimeEnvironmentExtended, IDeploymentParams, IDeploymentResult } from './types';
+import { constants } from 'ethers';
+import { requestDeployer } from '../src/lib';
+
+const AMOUNT = 0;
+
+export const xdeploy = async (
+  deploymentParams: IDeploymentParams,
+  hre: HardhatRuntimeEnvironmentExtended,
+): Promise<Array<IDeploymentResult>> => {
+  const { contract, constructorArgs } = deploymentParams;
+  console.log(`Deployment of ${contract} through xdeployer starting now !`);
+
+  await hre.run('compile');
+
+  if (!hre.config.xdeploy.networks || !hre.config.xdeploy.rpcUrls) {
+    throw new Error('Bad network configuration');
+  }
+
+  if (hre.config.xdeploy.networks.length !== hre.config.xdeploy.rpcUrls.length) {
+    throw new Error('Number of rpc url does not match number of network');
+  }
+
+  if (!hre.config.xdeploy.salt) {
+    throw new Error('Missing salt');
+  }
+
+  if (!hre.config.xdeploy.deployerAddress) {
+    throw new Error('Missing deployer address');
+  }
+
+  const providers: Array<any> = [];
+  const wallets: Array<any> = [];
+  const signers: Array<any> = [];
+  const create2Deployer: Array<any> = [];
+  const createReceipt: Array<any> = [];
```

Comment:
I'm pretty sure all these arrays should be replaced by unary constants set in the for loop